lib/mix/tasks/burn.ex

```diff
@@ -157,4 +145,58 @@ defmodule Mix.Tasks.Burn do
         Nerves.Env.firmware_path()
     end
   end
+
+  defp get_os!() do
+    case :os.type() do
+      {_, :linux} ->
+        if WSL.running_on_wsl?(), do: :wsl, else: :linux
+
+      {_, os} when os in [:darwin, :nt] ->
+        os
+
+      {_, os} ->
+        raise "Unable to burn firmware on your host #{inspect(os)}"
+    end
+  end
+
+  defp cmd_and_args_for_os(:linux, args, dev) do
+    fwup = System.find_executable("fwup")
+
+    case File.stat(dev) do
+      {:ok, %File.Stat{access: :read_write}} ->
+        {"fwup", args}
+
+      {:error, :enoent} ->
+        case File.touch(dev, System.os_time(:second)) do
+          :ok ->
+            {"fwup", args}
+
+          {:error, :eacces} ->
+            elevate_user()
+            {"sudo", provision_env() ++ [fwup] ++ args}
+        end
+
+      _ ->
+        elevate_user()
+        {"sudo", provision_env() ++ [fwup] ++ args}
+    end
+  end
+
+  defp cmd_and_args_for_os(:wsl, args, _dev) do
+    WSL.admin_powershell_command("fwup", Enum.join(args, " "))
+  end
+
+  defp cmd_and_args_for_os(_os, args, _dev), do: {"fwup", args}
+
+  defp format_result({_, 0}, _task), do: :ok
+
+  defp format_result({%FwupStream{output: o}, _}, "upgrade") do
+    if o =~ ~r/fwup: Expecting platform=#{mix_target()} and/ do
```

Comment:
This messages comes from whatever the user puts in their `fwup.config`, so it seems fragile. Does this work with a FAT-formatted MicroSD card? Like one that you get fresh from a store?

Reply:
Hmm, didn't think about that. I will try with a fresh card shortly

Reply:
I am getting the same message on a newly FAT formatted microSD and USB drive.

I'm wondering if they change here is to not be automatic and just prompt again asking for a confirmation to run the `complete` task? Or...we just always run complete if the `upgrade` task fails?

Reply:
Technically, even the task can be defined by the user as well and there may not even be an `upgrade` or `complete` task. Maybe this is considered a pure convenience and the only support for this "auto" work is with what's defined in official systems which all have this message and these tasks?